### PR TITLE
Adding px4 fmu v6c

### PR DESCRIPTION
**Pull request by davids5**
~~Untested waiting on HW~~

~~Need all sensors rotations~~

Tested on HW. 

1) @vincentpoont2 The reason the px4io is failing is the the units were shipped with 25.00 mHz Xatal at X7 this needs to be 24.00 mHz

To do: 

1) Need to verify sensors and Calibration - Done
2) Verify all signals on all connectors. - - Done

### CODE REVIEW

**Comment by davids5** [comment]
FYI @vincentpoont2

**Comment by vincentpoont2** [comment]
6C IMU

**Comment by vincentpoont2** [comment]
@davids5 we having this issue for all 3 units we tested

[boot] Rev 0x0 : Ver 0x0 V6C00
reset done, 10 ms
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
HW arch: PX4_FMU_V6C
HW type: V6C00
HW version: 0x00000000
HW revision: 0x00000000
FW git-hash: d5f5846d03ae23f91aacfe6feadf810138e325d2
FW version: 1.13.0 80 (17629312)
OS: NuttX
OS version: Release 10.2.0 (167903487)
OS git-hash: c5c7d2b4f26f52f1dfb425ebde83328606b65d4f
Build datetime: Apr 28 2022 17:37:16
Build uri: localhost
Build variant: default
Toolchain: GNU GCC, 9.3.1 20200408 (release)
PX4GUID: 000600000000353533373139510800480030
MCU: STM32H7[4|5]xxx, rev. V
[hardfault_log] Fault Log is Armed
INFO  [param] selected parameter default file /fs/mtd_params
INFO  [param] importing from '/fs/mtd_params'
INFO  [parameters] BSON document size 749 bytes, decoded 749 bytes (INT32:13, FLOAT:21)
INFO  [param] selected parameter backup file /fs/microsd/parameters_backup.bson
Board architecture defaults: /etc/init.d/rc.board_arch_defaults
Board defaults: /etc/init.d/rc.board_defaults
ERROR [param] Parameter MAV_2_BROADCAST not found.
ERROR [param] Parameter MAV_2_REMOTE_PRT not found.
ERROR [param] Parameter MAV_2_UDP_PRT not found.
nsh: safety_button: command not found
INFO  [dataman] data manager file '/fs/microsd/dataman' size is 62560 bytes
Loading airframe: /etc/init.d/airframes/4019_x500_v2
INFO  [uavcan] Node ID 1, bitrate 1000000
WARN  [px4io] check CRC failed: -1, CRC: 1891365525
WARN  [px4io] IO refused to re-enable safety
WARN  [px4io] IO refused to reboot
WARN  [px4io] reboot failed - -1, still attempting upgrade
[PX4IO] using firmware from /etc/extras/px4_io-v2_default.bin
[PX4IO] bad sync 0xff,0xff
[PX4IO] found bootloader revision: 5
[PX4IO] erase...
[PX4IO] programming 39952 bytes...
[PX4IO] verify...
[PX4IO] update complete
INFO  [tune_control] Stopping playback...
WARN  [px4io] check CRC failed: -1, CRC: 1891365525
PX4IO not found
Board sensors: /etc/init.d/rc.board_sensors
nsh: bmi055: command not found
nsh: bmi055: command not found
icm42688p #0 on SPI bus 1 rotation 6
ERROR [SPI_I2C] ms5611: no instance started (no device on bus?)
ERROR [SPI_I2C] ist8310: no instance started (no device on bus?)
ERROR [SPI_I2C] ist8310: no instance started (no device on bus?)
nsh: battery_status: command not found
ERROR [mixer] can't open /dev/pwm_output0

ERROR [mixer] failed to load mixer
ERROR  [init] Failed loading mixer: /etc/mixers/quad_x.main.mix
INFO  [init] setting PWM_AUX_OUT none
ekf2 [639:237]
Starting Main GPS on /dev/ttyS0
Starting MAVLink on /dev/ttyS5
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS5 @ 57600B
nsh: rc_input: command not found
INFO  [logger] logger started (mode=all)

**Comment by davids5** [comment]
@vincentpoont2 - sensors should work now. Please retest

**Comment by vincentpoont2** [comment]
@davids5 
Still having this issue with the latest Artifact on the 6C

[boot] Rev 0x0 : Ver 0x0 V6C00
reset done, 10 ms
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
HW arch: PX4_FMU_V6C
HW type: V6C00
HW version: 0x00000000
HW revision: 0x00000000
FW git-hash: 8c34ffb83af774db90396ca0ed4178c8d3df45cd
FW version: 1.13.0 80 (17629312)
OS: NuttX
OS version: Release 10.2.0 (167903487)
OS git-hash: c5c7d2b4f26f52f1dfb425ebde83328606b65d4f
Build datetime: Apr 29 2022 13:42:51
Build uri: localhost
Build variant: default
Toolchain: GNU GCC, 9.3.1 20200408 (release)
PX4GUID: 000600000000353533373139510800410028
MCU: STM32H7[4|5]xxx, rev. V
rdfault_log] Fault Log is Armed
INFO  [param] selected parameter default file /fs/mtd_params
INFO  [param] importing from '/fs/mtd_params'
INFO  [parameters] BSON document size 115 bytes, decoded 115 bytes (INT32:2, FLOAT:3)
INFO  [param] selected parameter backup file /fs/microsd/parameters_backup.bson
Board architecture defaults: /etc/init.d/rc.board_arch_defaults
Board defaults: /etc/init.d/rc.board_defaults
ERROR [param] Parameter MAV_2_BROADCAST not found.
ERROR [param] Parameter MAV_2_REMOTE_PRT not found.
ERROR [param] Parameter MAV_2_UDP_PRT not found.
nsh: safety_button: command not found
INFO  [dataman] data manager file '/fs/microsd/dataman' size is 62560 bytes
INFO  [uavcan] Node ID 1, bitrate 1000000
WARN  [px4io] check CRC failed: -1, CRC: 1891365525
WARN  [px4io] IO refused to re-enable safety
WARN  [px4io] IO refused to reboot
WARN  [px4io] reboot failed - -1, still attempting upgrade
[PX4IO] using firmware from /etc/extras/px4_io-v2_default.bin
[PX4IO] bad sync 0xff,0xff
[PX4IO] found bootloader revision: 5
[PX4IO] erase...
[PX4IO] programming 39952 bytes...
[PX4IO] verify...
[PX4IO] update complete
INFO  [tune_control] Stopping playback...
WARN  [px4io] check CRC failed: -1, CRC: 1891365525
PX4IO not found
Board sensors: /etc/init.d/rc.board_sensors
ERROR [SPI_I2C] bmi055: no instance started (no device on bus?)
ERROR [SPI_I2C] bmi055: no instance started (no device on bus?)
icm42688p #0 on SPI bus 1 rotation 6
ERROR [SPI_I2C] ms5611: no instance started (no device on bus?)
ist8310 #0 on I2C bus 4 address 0xC rotation 6
ERROR [SPI_I2C] ist8310: no instance started (no device on bus?)
nsh: icm20948_i2c_passthrough: command not found
nsh: battery_status: command not found
No autostart ID found
ekf2 [380:237]
Starting Main GPS on /dev/ttyS0
Starting MAVLink on /dev/ttyS5
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS5 @ 57600B
nsh: rc_input: command not found
INFO  [logger] logger started (mode=all)

NuttShell (NSH) NuttX-10.2.0
nsh>

**Comment by davids5** [comment]
@vincentpoont2 There are issues with the IMU board. I will contact you on Slack

**Comment by davids5** [comment]
```
NuttShell (NSH) NuttX-10.2.0
nsh> [boot] Rev 0x0 : Ver 0x0 V6C00
reset done, 10 ms
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
HW arch: PX4_FMU_V6C
HW type: V6C00
HW version: 0x00000000
HW revision: 0x00000000
FW git-hash: ffa1045eb858eac774094ee64a41a178e39a1bcd
FW version: 1.13.0 80 (17629312)
sercon: Registering CDC/ACM serial driver
FW git-branch: pr-px4_fmu-v6c
sercon: Successfully registered the CDC/ACM serial driver
OS: NuttX
OS version: Release 10.2.0 (167903487)
OS git-hash: c5c7d2b4f26f52f1dfb425ebde83328606b65d4f
Build datetime: May  6 2022 13:30:27
Build uri: localhost
Build variant: default
Toolchain: GNU GCC, 9.3.1 20200408 (release)
PX4GUID: 0006000000003535333731395108004a002c
MCU: STM32H7[4|5]xxx, rev. V
nsh: mount: mount failed: No such device
INFO [init] formatting /dev/mmcsd0
nsh: mkfatfs: mkfatfs failed: No such device
ERROR [init] format failed
INFO  [param] selected parameter default file /fs/mtd_params
INFO  [param] importing from '/fs/mtd_params'
INFO  [parameters] BSON document size 363 bytes, decoded 363 bytes (INT32:8, FLOAT:9)
Board architecture defaults: /etc/init.d/rc.board_arch_defaults
Board defaults: /etc/init.d/rc.board_defaults
ERROR [param] Parameter MAV_2_BROADCAST not found.
ERROR [param] Parameter MAV_2_REMOTE_PRT not found.
ERROR [param] Parameter MAV_2_UDP_PRT not found.
nsh: safety_button: command not found
WARN  [dataman] Could not open data manager file /fs/microsd/dataman
ERROR [dataman] dataman start failed
INFO  [uavcan] Node ID 1, bitrate 1000000
ERROR [uavcan] CentralizedServer init: -1
ERROR [uavcan] UavcanServers init: 0
INFO  [px4io] IO FW CRC match
Board sensors: /etc/init.d/rc.board_sensors
bmi055_accel #0 on SPI bus 1 rotation 4
bmi055_gyro #0 on SPI bus 1 rotation 4
icm42688p #0 on SPI bus 1 rotation 6
ms5611 #0 on I2C bus 4 address 0x77
ist8310 #0 on I2C bus 4 address 0xC rotation 6
ERROR [SPI_I2C] ist8310: no instance started (no device on bus?)
nsh: icm20948_i2c_passthrough: command not found
nsh: battery_status: command not found
No autostart ID found
ekf2 [335:237]
Starting Main GPS on /dev/ttyS0
Starting MAVLink on /dev/ttyS5
ERROR [mavlink] DM_KEY_MISSION_STATE lock failed
WARN  [mavlink] offboard mission init failed (-1)
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS5 @ 57600B
INFO  [logger] logger started (mode=all)
INFO  [logger] log root dir created: /fs/microsd/log

NuttShell (NSH) NuttX-10.2.0
nsh>
```

**Comment by vincentpoont2** [comment]
With the 24Mhz Xtal installed & the latest firmware artifact. BMI055 can be recognized. The battery voltage test failed.
Bootloader shows FMUV6X on QGC, assuming because it has the same PID. 
[20220509 FMUv6C print.txt](https://github.com/PX4/PX4-Autopilot/files/8650975/20220509.FMUv6C.print.txt)

> [boot] Rev 0x0 : Ver 0x0 V6C00
reset done, 10 ms
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
HW arch: PX4_FMU_V6C
HW type: V6C00
HW version: 0x00000000
HW revision: 0x00000000
FW git-hash: 1aed5a434da9d95a93a4eeeb57399a8329a3c2b1
FW version: 1.13.0 80 (17629312)
OS: NuttX
OS version: Release 10.2.0 (167903487)
OS git-hash: c5c7d2b4f26f52f1dfb425ebde83328606b65d4f
Build datetime: May  6 2022 23:33:30
Build uri: localhost
Build variant: default
Toolchain: GNU GCC, 9.3.1 20200408 (release)
PX4GUID: 0006000000003535333731395108003e0039
MCU: STM32H7[4|5]xxx, rev. V
[hardfault_log] Fault Log is Armed
INFO  [param] selected parameter default file /fs/mtd_params
INFO  [param] importing from '/fs/mtd_params'
INFO  [parameters] BSON document size 192 bytes, decoded 192 bytes (INT32:5, FLOAT:4)
INFO  [param] selected parameter backup file /fs/microsd/parameters_backup.bson
Board architecture defaults: /etc/init.d/rc.board_arch_defaults
Board defaults: /etc/init.d/rc.board_defaults
ERROR [param] Parameter MAV_2_BROADCAST not found.
ERROR [param] Parameter MAV_2_REMOTE_PRT not found.
ERROR [param] Parameter MAV_2_UDP_PRT not found.
nsh: safety_button: command not found
INFO  [dataman] data manager file '/fs/microsd/dataman' size is 62560 bytes
rgbled_ncp5623c #0 on I2C bus 1 (external) address 0x39
Loading airframe: /etc/init.d/airframes/4015_holybro_s500
INFO  [uavcan] Node ID 1, bitrate 1000000
INFO  [px4io] IO FW CRC match
Board sensors: /etc/init.d/rc.board_sensors
bmi055_accel #0 on SPI bus 1 rotation 4
bmi055_gyro #0 on SPI bus 1 rotation 4
icm42688p #0 on SPI bus 1 rotation 6
ms5611 #0 on I2C bus 4 address 0x77
ist8310 #0 on I2C bus 4 address 0xC rotation 6
ist8310 #1 on I2C bus 1 (external) address 0xE rotation 10
nsh: icm20948_i2c_passthrough: command not found
WARN  [SPI_I2C] Already running on bus 1
nsh: battery_status: command not found
INFO  [pwm_out] instance: 0, max rate: 100, default: 50, alt: 50
INFO  [init] Mixer: /etc/mixers/quad_x.main.mix on /dev/pwm_output0
INFO  [init] Mixer: /etc/mixers/pass.aux.mix on /dev/pwm_output1
ekf2 [657:237]
INFO  [pwm_out] instance: 0, max rate: 100, default: 50, alt: 50
Starting Main GPS on /dev/ttyS0
Starting MAVLink on /dev/ttyS5
WARN  [mavlink] offboard mission init failed (-1)
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS5 @ 57600B
INFO  [logger] logger started (mode=all)

NuttShell (NSH) NuttX-10.2.0
nsh> INFO  [gps] u-blox firmware version: SPG 3.01
INFO  [gps] u-blox protocol version: 18.00
INFO  [gps] u-blox module: NEO-M8N-0
INFO  [gps] u-blox firmware version: SPG 3.01
INFO  [gps] u-blox protocol version: 18.00
INFO  [gps] u-blox module: NEO-M8N-0

**Comment by vincentpoont2** [comment]
Possible to temporarily add some test software to the firmware, such as SD_bench, and enable it to print out the temperature of the CPU? It would help with our CPU temp testing.

**Comment by davids5** [comment]
@vincentpoont2 - 

USB ID and Board ID updated. 
Battery issues should be resolved 
Test Code added. 

For temperature see adc_report channel 20 and the datasheet for the conversion.
```
nsh> listener adc_report

TOPIC: adc_report
 adc_report
    timestamp: 541120274 (0.003151 seconds ago)
    device_id: 4294967295 (Type: 0xFF, UNKNOWN:31 (0xFF))
    raw_data: [0, 45, 17343, 175, 50442, 12964, 0, 0, 0, 0, 0, 0]
    resolution: 65536
    v_ref: 3.3000
    channel_id: [4, 5, 8, 14, 18, 20, -1, -1, -1, -1, -1, -1]
```

**Comment by vincentpoont2** [comment]
@davids5 
First indoor flight with the FMUv6C Hardware
https://logs.px4.io/plot_app?log=7296ea9a-f9ca-4400-a36a-df0bb474efe9

**Comment by davids5** [comment]
@vincentpoont2 - I fixed a bug that periph_5v_oc was reported wrong. Also the Safety switch CBRK default was not allowing the safety switch to be used. The board is added to the build archive now as well.

I have tested all signals at the connectors. It is now 100% functional. 

@dagar - this is ready to come in.

**Review by dagar — DISMISSED** [review]

File: boards/px4/fmu-v6c/init/rc.board_defaults [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Mavlink ethernet (CFG 1000)
+param set-default MAV_2_CONFIG 1000
+param set-default MAV_2_BROADCAST 1
+param set-default MAV_2_MODE 0
+param set-default MAV_2_RADIO_CTL 0
+param set-default MAV_2_RATE 100000
+param set-default MAV_2_REMOTE_PRT 14550
+param set-default MAV_2_UDP_PRT 14550
+
+param set-default CBRK_IO_SAFETY 0
```
Comment (dagar) [inline boards/px4/fmu-v6c/init/rc.board_defaults:22]: Disabling this by default was intentional, most people don't actually want to use it.

Comment (dagar) [inline boards/px4/fmu-v6c/init/rc.board_defaults:22]:
```suggestion
```

**Comment by vincentpoont2** [comment]
@davids5 
We placed the IST8310 sensor to a position far away from the flight control, but still did not solve the problem of unstable positioning. The external IST8310 has been connected to other flight controllers for calibration, and it is confirmed that the OFFSET of the XYZ axis is not greater than 200. The possibility of a bad compass was ruled out.

The test flight Log here [https://review.px4.io/plot_app?log=2c21a087-7159-4e2d-b0f0-3cae9812ea47](https://review.px4.io/plot_app?log=2c21a087-7159-4e2d-b0f0-3cae9812ea47)

**Comment by davids5** [comment]
> @davids5 We placed the IST8310 sensor to a position far away from the flight control, but still did not solve the problem of unstable positioning. The external IST8310 has been connected to other flight controllers for calibration, and it is confirmed that the OFFSET of the XYZ axis is not greater than 200. The possibility of a bad compass was ruled out.
> 
> The test flight Log here https://review.px4.io/plot_app?log=2c21a087-7159-4e2d-b0f0-3cae9812ea47

@vincentpoont2 - see slack.

We will bring this is and continue to debug the issue.

**Comment by dagar** [comment]
There's something wrong with the internal IST8310. In both David's board and Vince's test flight log the internal IST8310 X axis is reporting values that are at least an order of magnitude too large.
Screenshot from 2022-05-20 15-47-37

I'm going to implement the IST8310 test procedure in the driver startup and we'll continue debugging.

**Review by dagar — APPROVED** [review]

**Pull request merged**